### ISSUE TITLE
Building Blaster - Long Loading Screen than Black Screen Error

### ISSUE THREAD

**Issue by desertking** [ruffle-rs/ruffle#8536]
### Describe the bug

Hey,

I guess the first error has to do with the ad-preloader, maybe some timeout but it loads after like 20 second then the intro shows up but it doesn't get behind that and an error is thrown.

https://www.newgrounds.com/portal/view/508522?emulate=flash

0184931e:0x4b6bef Error dispatching event EventObject(EventObject { type: "ApplicationEvent.ready", class: com.bigwigmedia.events::ApplicationEvent, ptr: 0x12916e0 }) to handler FunctionObject(FunctionObject(GcCell(Gc { ptr: 0x6f5fac }))) : RustError("Non-public property flash.events:IEventDispatcher::addEventListener not found on Object")

### Expected behavior

Start the game..

### Affected platform

Self-hosted version

### Operating system

All

**Comment by n0samu** [comment]
Interestingly, a different bug occurs if the SWF's network requests are not blocked by CORS. In that case, Ruffle tries to load the ad-preloader (which is still online) and this happens:
```
[WARN  ruffle_core::display_object::movie_clip] DoAction tag in AVM2 movie
thread 'main' panicked at 'No script object for display object', core\src\avm1\runtime.rs:167:18
```

**Comment by Lord-McSweeney** [comment]
The ad preloader is no longer online, so to run the game, you either have to wait for the request to time out or turn off internet.

This needs XML coercion:
```
ERROR core/src/avm2/events.rs:419 Error dispatching event EventObject(EventObject { type: "ApplicationEvent.ready", class: com.bigwigmedia.events::ApplicationEvent, ptr: 0x22f04a8 }) to handler FunctionObject(FunctionObject { ptr: 0x80c0a8 }) : RustError("Cannot coerce Object(ByteArrayObject(ByteArrayObject { ptr: 0x2337a40 })) to an XML")
```

**Comment by Lord-McSweeney** [comment]
On latest nightly, this needs `Graphics.beginBitmapFill`. Level numbering text at the level select is also moved incorrectly.

**Comment by n0samu** [comment]
Should test with #10123 if anyone has a chance

**Comment by n0samu** [comment]
Fixed by #10123. (Text issues still exist but can be addressed separately)
